src/proto/Cosmos.proto

```diff
@@ -205,6 +205,62 @@ message Message {
         bytes auth_info_bytes = 2; 
     }
 
+    // StakeAuthorization defines authorization for delegate/undelegate/redelegate.
+    //
+    // Since: cosmos-sdk 0.43
+    message StakeAuthorization {
+        // max_tokens specifies the maximum amount of tokens can be delegate to a validator. If it is
+        // empty, there is no spend limit and any amount of coins can be delegated.
+        Amount max_tokens = 1;
+        // validators is the oneof that represents either allow_list or deny_list
+        oneof validators {
+            // allow_list specifies list of validator addresses to whom grantee can delegate tokens on behalf of granter's
+            // account.
+            Validators allow_list = 2;
+            // deny_list specifies list of validator addresses to whom grantee can not delegate tokens.
+            Validators deny_list = 3;
+        }
+        // Validators defines list of validator addresses.
+        message Validators {
+            repeated string address = 1;
+        }
+        // authorization_type defines one of AuthorizationType.
+        AuthorizationType authorization_type = 4;
+    }
+
+    // AuthorizationType defines the type of staking module authorization type
+    //
+    // Since: cosmos-sdk 0.43
+    enum AuthorizationType {
+        // AUTHORIZATION_TYPE_UNSPECIFIED specifies an unknown authorization type
+        AUTHORIZATION_TYPE_UNSPECIFIED = 0;
+        // AUTHORIZATION_TYPE_DELEGATE defines an authorization type for Msg/Delegate
+        AUTHORIZATION_TYPE_DELEGATE = 1;
+        // AUTHORIZATION_TYPE_UNDELEGATE defines an authorization type for Msg/Undelegate
+        AUTHORIZATION_TYPE_UNDELEGATE = 2;
+        // AUTHORIZATION_TYPE_REDELEGATE defines an authorization type for Msg/BeginRedelegate
+        AUTHORIZATION_TYPE_REDELEGATE = 3;
+    }
+
+
+    // cosmos-sdk/MsgGrant
+    message AuthGrant {
+        string granter = 1;
+        string grantee = 2;
+        oneof grant_type {
+            StakeAuthorization grant_stake = 3;
+        }
+        int64 expiration = 4;
+        string type_url = 5;
+    }
+
+    // cosmos-sdk/MsgRevoke
+    message AuthRevoke {
+        string granter      = 1;
+        string grantee      = 2;
+        string msg_type_url = 3;
```

Comment:
same here, looks implementation details to me, trying to hide as much as possible

Reply:
Same answer as above `msg_type_url` is required to know which kind of **authz** we are revoking